internal/storage/db.go

```diff
@@ -76,3 +82,39 @@ func (s *DBStorage) AssignTokens(ctx context.Context, issuanceWeek, firstAutomat
 	_, err = s.DBS.DBS().Writer.ExecContext(ctx, assignTokensQuery, issuanceWeek, weekLimit.String())
 	return err
 }
+
+var tokensPerWeekQuery = `
+SELECT 
+	SUM(r.tokens)::numeric / NULLIF(SUM(
+        r.streak_points + 
+        r.integration_points + 
+        r.aftermarket_device_points + 
+        r.synthetic_device_points
+    ), 0) * $2 as tokens_for_points
+FROM rewards_api.issuance_weeks w
+JOIN rewards_api.rewards r TABLESAMPLE BERNOULLI(10)
```

Comment:
This makes me nervous. I think sampling happens before where conditions are invoked.

Reply:
True I just like sampling because it was 30ms when using limit 10 with no order and 4ms for sampling 10%

Reply:
What abouut